subprojects/platform-native/src/main/java/org/gradle/nativeplatform/tasks/AbstractLinkTask.java

```diff
@@ -35,105 +39,94 @@
 import org.gradle.internal.Cast;
 import org.gradle.internal.operations.logging.BuildOperationLogger;
 import org.gradle.internal.operations.logging.BuildOperationLoggerFactory;
+import org.gradle.language.base.compile.CompilerVersion;
 import org.gradle.language.base.internal.compile.Compiler;
+import org.gradle.language.base.internal.compile.VersionAwareCompiler;
 import org.gradle.language.base.internal.tasks.SimpleStaleClassCleaner;
 import org.gradle.nativeplatform.internal.BuildOperationLoggingCompilerDecorator;
 import org.gradle.nativeplatform.internal.LinkerSpec;
 import org.gradle.nativeplatform.platform.NativePlatform;
 import org.gradle.nativeplatform.platform.internal.NativePlatformInternal;
 import org.gradle.nativeplatform.toolchain.NativeToolChain;
 import org.gradle.nativeplatform.toolchain.internal.NativeToolChainInternal;
+import org.gradle.nativeplatform.toolchain.internal.PlatformToolProvider;
 
 import javax.inject.Inject;
-import java.io.File;
-import java.util.concurrent.Callable;
 
 /**
  * Base task for linking a native binary from object files and libraries.
  */
 @Incubating
 public abstract class AbstractLinkTask extends DefaultTask implements ObjectFilesToBinary {
-    private NativeToolChainInternal toolChain;
-    private NativePlatformInternal targetPlatform;
-    private boolean debuggable;
-    private final RegularFileProperty outputFile;
+    private final RegularFileProperty linkedFile;
+    private final DirectoryProperty destinationDirectory;
     private final ListProperty<String> linkerArgs;
     private final ConfigurableFileCollection source;
     private final ConfigurableFileCollection libs;
+    private final Property<Boolean> debuggable;
+    private final Property<NativePlatform> targetPlatform;
+    private final Property<NativeToolChain> toolChain;
 
     public AbstractLinkTask() {
-        libs = getProject().files();
-        source = getProject().files();
-        outputFile = newOutputFile();
-        linkerArgs = getProject().getObjects().listProperty(String.class);
-        getInputs().property("outputType", new Callable<String>() {
```

Comment:
❌ Should this really be removed?

Reply:
Yep, this was tracking the implementation of the linker and I just copied what we were doing in the compilation tasks now.